app/scripts/views/settings/communication_preferences.js

```diff
@@ -93,35 +94,35 @@ define(function (require, exports, module) {
       return this.setOptInStatus(NEWSLETTER_ID, ! emailPrefs.isOptedIn(NEWSLETTER_ID));
     },
 
-    setOptInStatus (newsletterId, isOptedIn) {
-      let eventPrefix;
-      let method;
+    setOptInStatus (newsletterId, isOptingIn) {
+      const emailPrefs = this.getMarketingEmailPrefs();
 
-      if (isOptedIn) {
-        eventPrefix = '';
-        method = 'optIn';
-      } else {
-        eventPrefix = 'un';
-        method = 'optOut';
-      }
+      if (isOptingIn) {
+        this.logViewEvent('optIn');
 
-      this.logViewEvent(method);
+        return emailPrefs.optIn(newsletterId)
+          .then(() => {
+            this.logViewEvent('optIn.success');
+            // Emit an additional flow event for consistency with
+            // the call to optIn in the account model
+            this.logFlowEvent('newsletter.subscribed');
+            this.displaySuccess(t('Subscribed successfully'));
+            this.navigate('settings');
+            return this.render();
+          }, (err) => this.displayError(err));
 
-      return this.getMarketingEmailPrefs()[method](newsletterId)
-        .then(() => {
-          this.logViewEvent(method + '.success');
-          // Emit an additional flow event for consistency with
-          // the call to optIn in the account model
-          this.logFlowEvent(`newsletter.${eventPrefix}subscribed`);
+      } else {
+        return p().then(() => {
+          // preferencesURL is only available if the user is already
+          // registered with basket.
+          this.logViewEvent('manage');
+          this.logFlowEvent('newsletter.manage');
+
+          return this.window.open(emailPrefs.get('preferencesUrl'), '_blank');
```

Comment:
Like I said above, it seems like we should be able to use an anchor element from the HTML to open the preferencesUrl instead of using JS to do so. 

I tried and it looks like the CSS isn't set up to make an anchor look like a button, but I didn't look very hard. Not a blocker, just seems like the semantic thing to do.

Reply:
While testing, `preferencesUrl` is always `null` which causes /settings/communication_preferences to be opened. This is because locally, we don't have `preferencesUrl` because [of the lack of SQS](https://github.com/mozilla/fxa-content-server/issues/5433).